README.md

```diff
@@ -38,3 +38,197 @@ Then start the application locally:
+
+        except Exception as e:
+            print(f"Failed to load {url} ({title}): {e}")
+
+```
+3. To enable new plugin, add it to settings_override.py:
+```python
+CATALOG_DOCUMENT_SOURCE_PLUGINS = {
+    "PDF Plugin": "<path_to_file>.PDFDocumentSourcePlugin"
+}
+```
+### Next, let's move to agent. 
+1. Create a directory for you agent (e.g. `pdf_agent`). Then inside it create `agent.py` file:
+```python
+from enthusiast_common.agents import BaseAgent
+from enthusiast_common.injectors import BaseInjector
+from enthusiast_common.tools.base import BaseTool
+from langchain.agents import AgentExecutor, create_tool_calling_agent
+from langchain_core.callbacks import BaseCallbackHandler
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.prompts import ChatPromptTemplate
+
+
+class ExamplePDFAgent(BaseAgent):
+    def __init__(
+        self,
+        tools: list[BaseTool],
+        llm: BaseLanguageModel,
+        prompt: ChatPromptTemplate,
+        conversation_id: int,
+        injector: BaseInjector,
+        callback_handler: BaseCallbackHandler | None = None,
+    ):
+        super().__init__(
+            tools=tools,
+            llm=llm,
+            prompt=prompt,
+            conversation_id=conversation_id,
+            callback_handler=callback_handler,
+            injector=injector,
+        )
+        self._agent_executor = self._create_agent_executor()
+
+    def _create_agent_executor(self, **kwargs) -> AgentExecutor:
+        tools = self._create_tools()
+        agent = create_tool_calling_agent(self._llm, tools, self._prompt)
+        return AgentExecutor(
+            agent=agent, tools=tools, verbose=True, memory=self._injector.chat_summary_memory, **kwargs
+        )
+
+    def _create_tools(self):
+        return [tool_class.as_tool() for tool_class in self._tools]
+
+    def get_answer(self, input_text: str) -> str:
+        agent_output = self._agent_executor.invoke(
+            {"input": input_text}, config={"callbacks": [self._callback_handler] if self._callback_handler else []}
+        )
+        return agent_output["output"]
+```
+2. Create Prompt in prompt.py file.
+```python
+PDF_AGENT_SYSTEM_PROMPT="""
+You are a helpful agent, answering questions about pdf document. Always use context tool
+"""
+```
+
+3. Create context retrieving tool:
+```python
+import tiktoken
+from enthusiast_common.injectors import BaseInjector
+from enthusiast_common.tools import BaseLLMTool
+from langchain_core.language_models import BaseLanguageModel
+from pydantic import BaseModel, Field
+
+
+class ContextSearchToolInput(BaseModel):
+    full_user_request: str = Field(description="user's full request")
+
+
+class ContextSearchTool(BaseLLMTool):
+    NAME = "context_search_tool"
+    DESCRIPTION = "Use it to get context from pdf required for answering questions"
+    ARGS_SCHEMA = ContextSearchToolInput
+    RETURN_DIRECT = False
+
+    ENCODING: tiktoken.encoding_for_model = None
```

Comment:
We don't need the ENCODING here, I think we could get rid of it and simplify the agent